auth-krb5.c

```diff
@@ -242,11 +244,14 @@ krb5_cleanup_proc(Authctxt *authctxt)
 krb5_error_code
 ssh_krb5_cc_gen(krb5_context ctx, krb5_ccache *ccache) {
 	int tmpfd, ret, oerrno;
-	char ccname[40];
+	char ccname[PATH_MAX] = {0};
+	char *path;
 	mode_t old_umask;
 
+	path = session_get_runtime_directory();
```

Comment:
I don't really want the auth code depending on the session code because it interferes with the goal of splitting auth-related things from the sshd process that handles user sessions. Can you adapt this to use the version you have in misc.c? Alternately an auth_get_runtime_directory() might be acceptable.